models/user.js

```diff
@@ -142,32 +146,65 @@ UserSchema.statics.authenticateSocial = async (req, res, next) => {
 };
 
 UserSchema.statics.isUserInDB = async (email, nickname) => {
-   let res = await User.find( { $or: [ { email: email }, { nickname: nickname } ] } );
-   return res.length;
+  let res = await User.find({ $or: [{ email: email }, { nickname: nickname }] });
+  return res.length;
 }
 
 UserSchema.statics.validate = (req, res, next) => {
   req.checkBody('nickname', 'Name is required').notEmpty();
   req.checkBody('email', 'Name is required').notEmpty()
   req.checkBody('email', 'Email must be valid').isEmail();
-  req.checkBody('passwordConf','Password is required').notEmpty()
-  req.checkBody('passwordConf','Password must contain at least 6 characters').isLength({ min:  6});
-  req.checkBody('password','Password is required').notEmpty();
-  req.checkBody('password','Password must contain at least 6 characters').isLength({ min:  6});
-  req.checkBody('password','Passwords do not matched').equals(req.body.passwordConf);
+  req.checkBody('passwordConf', 'Password is required').notEmpty()
+  req.checkBody('passwordConf', 'Password must contain at least 6 characters').isLength({ min: 6 });
+  req.checkBody('password', 'Password is required').notEmpty();
+  req.checkBody('password', 'Password must contain at least 6 characters').isLength({ min: 6 });
+  req.checkBody('password', 'Passwords do not matched').equals(req.body.passwordConf);
 
   let errors = req.validationErrors();
   if (errors) {
-    let err =  new Error (errors[0].msg);
+    let err = new Error(errors[0].msg);
     err.status = 422;
     return next(err);
   }
   return next();
 }
 
-UserSchema.pre('save',  function (next) {
+UserSchema.statics.userProfile = async (req, res, next) => {
+  let nickname = req.params.nickname;
+  let profile = await User.find({ nickname: nickname });
+  if (profile.length > 0) {
+    res.status(200).json({ userprofile: prepareUser(profile[0]) });
+  } else {
+    let error = new Error("User not found");
+    error.status = 404;
+    next(error);
+  }
+};
+
+UserSchema.statics.saveEditProfile = async (req, res, next) => {
+  let userId = req.params.id;
+  let profile = await User.find({ _id: userId });
+  if (profile.length > 0) {
+    User.update({ _id: userId },
+      {
+        $set: {
+          userName: req.body.userName,
+          nickname: req.body.nickname,
+          website: req.body.website,
+          userBio: req.body.userBio
+        }
+      }, function (err, userProfile) {
+        if (err) return next(err);
+        res.status(200).send({ userprofile: prepareUser(userProfile) });
+      });
+  } else {
+    res.status(404).send();
```

Comment:
you need create error object with status 404 and use next(err)

Reply:
let error = new Error ('Eror msg')
    error.status = 404;
    next(error);